app/models/notifications/mentioned.rb

```diff
@@ -13,19 +13,26 @@ def deleted_translation_key
     end
 
     def linked_object
-      target.post
+      target.mentions_container
     end
 
     def self.notify(mentionable, recipient_user_ids)
       actor = mentionable.author
+      relevant_mentions = filter_mentions(
+        mentionable.mentions.local.where.not(person: mentionable.author),
+        mentionable,
+        recipient_user_ids
+      )
 
-      mentionable.mentions.select {|mention| mention.person.local? }.each do |mention|
+      relevant_mentions.each do |mention|
         recipient = mention.person
-
-        next if recipient == actor || !(mentionable.public || recipient_user_ids.include?(recipient.owner_id))
-
         create_notification(recipient.owner_id, mention, actor).email_the_user(mention, actor)
       end
     end
+
+    def self.filter_mentions(mentions, mentionable, recipient_user_ids)
```

Comment:
Does it mean that I have to write a migration in order to change the `type` of notification in the DB  or can I just override the convention like this?
```ruby
class Notification::MentionedInPost < Notification
...
  def self.sti_name
      "Notification::Mentioned"
  end
...
end
```